### PR TITLE
Completed beforenm and open_afternm wrappers

### PR DESCRIPTION
**Pull request by mpcsh**
Just more wrappers. Fully tested. Made tests run a bit faster by using a 2MB message instead of 20.

### CODE REVIEW

File: src/sodium.iced [hunk 1]
```diff
@@ -97,4 +97,23 @@ exports.Sodium = class Sodium extends Base
...
+
+  #
+  # @method box_open_afternm
+  # Precopmute a shared secret
+  # @param {Buffer} ciphertext The ciphertext to decrypt
+  # @param {Buffer} nonce The nonce
+  # @param {Buffer} secret The precomputed secret
+  # @return {Buffer} The decrypted ciphertext
+  box_open_afternm : ({ciphertext, nonce, secret}) ->
+    return @lib.c.crypto_box_open_afternm(Buffer.concat([Buffer.alloc(16), ciphertext]), nonce, secret)
```
Comment (maxtaco) [inline src/sodium.iced]: can you factor out the zero-pad operation into a subroutine?

**Comment by maxtaco** [comment]
one small comment...

File: src/sodium.iced (hunk 1 above)
Comment (mpcsh) [inline src/sodium.iced]: ec8f4d1

**Comment by maxtaco** [comment]
👍

**Pull request merged**